### ISSUE TITLE
WP 5.6 performance degradation

### ISSUE THREAD

**Issue by peterfabian** [woocommerce/woocommerce#28140]
Investigate and fix WP 5.6 performance problem

**Comment by lkraav** [comment]
> Investigate and fix WP 5.6 performance problem

Are there any details in some other issue, or will this issue opening comment be updated at some later point?

**Comment by peterfabian** [comment]
@vedanshujain Can you please add more background here?

**Comment by vedanshujain** [comment]
I have added a [ticket in core trac](https://core.trac.wordpress.org/ticket/51678#ticket) with the necessary details and possible solution:

Paraphrasing to save a click:



In WooCommerce's internal autmated performance testing, we are seeing performance degradations in all paths where any plugin is loaded whose translation file is not present. Root cause seems to in [#49236](https://core.trac.wordpress.org/changeset/49236), where we added some changes and introduced WP_Textdomain_Registry to store text domains and their language directory paths.

Looks like, if translation file is not present for a domain, we hit the is_readable on every l10n call. While cached internally, it seems like this call is still slow on many hosts and it will be good to add additional caching around it.

Adding line $wp_textdomain_registry->set( $domain, false ); inside if ( ! is_readable( $mofile ) ) block restores the performance on our test sites, and I'd recommend that we commit this change, because:

1. It's unlikely that the result of is_readable( $mofile ) will ever change during the context of a request, so it should be fine to cache this result in wp_textdomain_registry. 

2. Setting it to false here will cause an early return inside _load_textdomain_just_in_time, thereby reducing calls to is_readaable eventually. 

3. As far as I can see, this still preserves the original fix intended in #49236. 

4. We were in fact caching the output of this call previous to #49236. 


---------

Let's close the issue when the core issue is resolved.

**Comment by peterfabian** [comment]
TL;DR update: there are several patches attached to the WP core ticket, but it seems like the code might be rolled back after all, as there isn't that much time to fix the problems and come up with a good tested solution until WP 5.6 lands. 

If the problem won't be solved, we have a strategy for a workaround by @vedanshujain :
> I want to hook into `load_textdomain_mofile` to set the domain to false in registry like I suggested in the ticket description, and then hook again into switch_to_locale function to clear this cache whenever locale changes to counter the locale not changing issue mentioned by Tonya

**Comment by vedanshujain** [comment]
This was reverted in WP core, marking this issue as closed, see https://core.trac.wordpress.org/changeset/49566
